src/iothread.cpp

```diff
@@ -199,7 +199,9 @@ static void enqueue_thread_result(work_request_t req) {
     s_result_queue.acquire()->push(std::move(req));
 }
 
-static void *this_thread() { return (void *)(intptr_t)pthread_self(); }
+static void *this_thread() {
+    return reinterpret_cast<void *>(static_cast<intptr_t>(pthread_self()));
```

Comment:
Breaks the build on macOS:

>/Users/travis/build/fish-shell/fish-shell/src/iothread.cpp:203:37: error: static_cast from 'pthread_t _Nonnull' (aka '_opaque_pthread_t *') to 'intptr_t' (aka 'long') is not allowed

>return reinterpret_cast<void *>(static_cast<intptr_t>(pthread_self()));

Reply:
Switched back to C cast for that since it's static on some platforms and reinterpret_cast on others.